payara-managed/src/main/java/fish/payara/arquillian/container/payara/managed/PayaraServerControl.java

```diff
@@ -222,8 +226,34 @@ private ProcessOutputConsumer createProcessOutputConsumer() {
         if (config.isOutputToConsole()) {
             return new OutputLoggingConsumer();
         }
-        
+
         return new SilentOutputConsumer();
     }
 
+    private File getJavaProgram() {
+        final File asjJavaCommand = getJavaProgramFromEnv("AS_JAVA");
+        if (asjJavaCommand != null) {
+            return asjJavaCommand;
+        }
+        final File javaHomeCommand = getJavaProgramFromEnv("JAVA_HOME");
+        if (javaHomeCommand != null) {
+            return javaHomeCommand;
+        }
+        return null;
+    }
+
+    private File getJavaProgramFromEnv(final String key) {
+        final String property = System.getenv(key);
+        if (property == null) {
+            return null;
+        }
+        final Path mainFolder = new File(property).toPath();
+        final File java = mainFolder.resolve("bin").resolve("java").toFile();
```

Comment:
For Windows OS, `java.exe` should also be checked.

Reply:
Aha, you are right, because previous version run java command, and windows adds the suffix automatically. But this would not work. I will fix it ...

Reply:
Fixed, thank you!